test/dataframe.jl

```diff
@@ -1976,4 +1976,89 @@ end
     @test select(df, Not([:x2, :x3]), All()) == select(df, :x1, :x4, :x2, :x3)
 end
 
+@testset "vcat and push! with :identical" begin
+    for v in (Dict(:a=>10, :b=>20, :c=>30), (a=10, b=20, c=30),
+              DataFrame(a=10, b=20, c=30)[1, :])
+        df = DataFrame(a=1, b=2, c=3)
+        push!(df, v, cols=:identical)
+        @test df == DataFrame(a=[1,10], b=[2,20], c=[3,30])
+    end
+
+    df = DataFrame(a=1, b=2, c=3)
+    v = Dict(:a=>10, :b=>20, :d=>30)
+
+    # avoid printing of error to the console during test
+    old_logger = global_logger(NullLogger())
+    @test_throws KeyError push!(df, v, cols=:identical)
+    global_logger(old_logger)
+
+    for v in ((a=10, b=20, d=30), (a=10, c=20, b=30),
+              DataFrame(a=10, c=20, b=30)[1, :], Dict(:a=>10, :b=>20, :c=>30, :d=>0),
+              (a=10, b=20, c=30, d=0), DataFrame(a=10, b=20, c=30, d=0)[1, :])
+        df = DataFrame(a=1, b=2, c=3)
+        @test_throws ArgumentError push!(df, v, cols=:identical)
+    end
+
+    @test vcat(DataFrame(a=1, b=2, c=3), DataFrame(a=10, b=20, c=30),
+               cols=:identical) == DataFrame(a=[1,10], b=[2,20], c=[3,30])
+    @test_throws ArgumentError vcat(DataFrame(a=1, b=2, c=3), DataFrame(a=10, c=20, b=30),
+                                    cols=:identical)
+    @test_throws ArgumentError vcat(DataFrame(a=1, b=2, c=3), DataFrame(a=10, b=20, d=30),
+                                    cols=:identical)
+    @test_throws ArgumentError vcat(DataFrame(a=1, b=2, c=3), DataFrame(a=10, b=20, c=30, d=0),
+                                    cols=:identical)
```

Comment:
Which logger is wanted here?

Reply:
to avoid printing error messages to the console

Reply:
Then we miss the NullLogger here. (The original logger is restored before this code snippet)

Reply:
But we don't need a logger for errors, right?

Reply:
We need a logger only to cover `@error`, which does not throw an exception, but prints an error to `stderr`. I will check if this is a case here.